src/cascadia/PublicTerminalCore/HwndTerminal.hpp

```diff
@@ -0,0 +1,61 @@
+#pragma once
+
+#include "../../renderer/base/Renderer.hpp"
+#include "../../renderer/dx/DxRenderer.hpp"
+#include "../../cascadia/TerminalCore/Terminal.hpp"
+
+extern "C" {
+    __declspec(dllexport) HRESULT _stdcall CreateTerminal(HWND parentHwnd, _Out_ void** hwnd, _Out_ void** terminal);
+    __declspec(dllexport) void _stdcall SendTerminalOutput(void* terminal, LPCWSTR data);
+    __declspec(dllexport) void _stdcall RegisterScrollCallback(void* terminal, void __stdcall callback(int, int, int));
+    __declspec(dllexport) HRESULT _stdcall TriggerResize(void* terminal, double width, double height, _Out_ int* charColumns, _Out_ int* charRows);
+    __declspec(dllexport) void _stdcall DpiChanged(void* terminal, int newDpi);
+    __declspec(dllexport) void _stdcall UserScroll(void* terminal, int viewTop);
+    __declspec(dllexport) void _stdcall StartSelection(void* terminal, COORD cursorPosition, bool altPressed);
+    __declspec(dllexport) void _stdcall MoveSelection(void* terminal, COORD cursorPosition);
+    __declspec(dllexport) void _stdcall ClearSelection(void* terminal);
+    __declspec(dllexport) const wchar_t* _stdcall GetSelection(void* terminal);
+    __declspec(dllexport) bool _stdcall IsSelectionActive(void* terminal);
+    __declspec(dllexport) void _stdcall DestroyTerminal(void* terminal);
+
+    enum CursorStyle
+    {
+
+    };
+    struct TerminalSettings
+    {
+        COLORREF DefaultBackground;

```

Comment:
I'm concerned that this is in no way tied to any other settings objects so it changes when they do.